class-gf-razorpay.php

```diff
@@ -0,0 +1,336 @@
+                        'class'   => 'medium',
+                    ),
+                    array(
+                        'type' => 'save',
+                        'messages' => array(
+                            'success' => esc_html__('Settings have been updated.', 'razorpay-gravity-forms' )
+                        ),
+                    ),
+                ),
+            ),
+        );
+    }
+
+    public function get_customer_fields_array($feed, $entry)
+    {
+        $fields = array();
+
+        $customerFields = $this->get_customer_fields();
+
+        foreach ($customerFields as $field)
+        {
+            $fieldId = $feed['meta'][$field['meta_name']];
+
+            $value = rgar($entry, $fieldId);
+
+            $fields[$field['meta_name']] = $value;
+        }
+
+        return $fields;
+    }
+
+    public function get_customer_fields()
+    {
+        return array(
+            array(
+                'name'      => 'first_name',
+                'label'     => 'First Name',
+                'meta_name' => 'billingInformation_firstName'
+            ),
+            array(
+                'name'      => 'last_name',
+                'label'     => 'Last Name',
+                'meta_name' => 'billingInformation_lastName'
+            ),
+            array(
+                'name'      => 'email',
+                'label'     => 'Email',
+                'meta_name' => 'billingInformation_email'
+            ),
+            array(
+                'name'      => 'address1',
+                'label'     => 'Address',
+                'meta_name' => 'billingInformation_address'
+            ),
+            array(
+                'name'      => 'address2',
+                'label'     => 'Address 2',
+                'meta_name' => 'billingInformation_address2'
+            ),
+            array(
+                'name'      => 'city',
+                'label'     => 'City',
+                'meta_name' => 'billingInformation_city'
+            ),
+            array(
+                'name'      => 'state',
+                'label'     => 'State',
+                'meta_name' => 'billingInformation_state'
+            ),
+            array(
+                'name'      => 'zip',
+                'label'     => 'Zip',
+                'meta_name' => 'billingInformation_zip'
+            ),
+            array(
+                'name'      => 'country',
+                'label'     => 'Country',
+                'meta_name'  => 'billingInformation_country'
+            ),
+        );
+    }
+
+    public function callback()
+    {
+        $entryId = $_COOKIE[self::RAZORPAY_ENTRY_ID];
+
+        $razorpayOrderId = $_COOKIE[self::RAZORPAY_ORDER_ID];
+
+        $entry = GFAPI::get_entry($entryId);
+
+        $key = $this->get_plugin_setting(self::GF_RAZORPAY_KEY);
+
+        $secret = $this->get_plugin_setting(self::GF_RAZORPAY_SECRET);
+
+        $api = new Api($key, $secret);
+
+        $attributes = array (
```

Comment:
@riagarg5 a cookie is considered to be equally tainted as user input as a post input parameter.